healthcheck_test.go

```diff
@@ -27,3 +31,28 @@ func TestCheck(t *testing.T) {
 	require.EqualError(t, errFailed, "failed to parse status JSON: invalid character '\\x01' looking for beginning of value")
 	require.EqualValues(t, "failed to parse status JSON: invalid character '\\x01' looking for beginning of value", statusFailed.Error)
 }
+
+func TestDumpToDisk(t *testing.T) {
+	successfulOutput := "./tmp/success.txt"
+	os.Remove(successfulOutput)
+
+	ctx, cancelSuccess := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancelSuccess()
+
+	DumpToDisk(ctx, successfulOutput, []byte("success"))
+
+	data, err := ioutil.ReadFile(successfulOutput)
+	require.NoError(t, err)
+	require.Equal(t, data, []byte("success"))
+
+	timedOutOutput := "./tmp/timed-out.txt"
+	os.Remove(timedOutOutput)
+
+	ctx, cancelTimeout := context.WithTimeout(context.Background(), 0)
```

Comment:
But in the test, you are testing the timeout when there is no timeout. the code is actually broken and the test does not detect it